pkg/bootstrap/os/tasks.go

```diff
@@ -454,3 +454,71 @@ func (u *UmountISO) Execute(runtime connector.Runtime) error {
 	}
 	return nil
 }
+
+type NodeConfigureNtpCheck struct {
+	common.KubePrepare
+}
+
+func (n *NodeConfigureNtpCheck) PreCheck(runtime connector.Runtime) (bool, error) {
+
+	if len(n.KubeConf.Cluster.System.NtpServers) == 0 && len(n.KubeConf.Cluster.System.Timezone) == 0 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+type NodeConfigureNtpServer struct {
+	common.KubeAction
+}
+
+func (n *NodeConfigureNtpServer) Execute(runtime connector.Runtime) error {
+
+	currentHost := runtime.RemoteHost()
+
+	for _, server := range n.KubeConf.Cluster.System.NtpServers {
+
+		serverAddr := strings.Trim(server, " \"")
+		if serverAddr == currentHost.GetName() || serverAddr == currentHost.GetInternalAddress() {
+			allowClientCmd := `sed -i '/#allow/ a\allow 0.0.0.0/0' /etc/chrony.conf`
+			if _, err := runtime.GetRunner().SudoCmd(allowClientCmd, false); err != nil {
+				return errors.Wrapf(err, "change host:%s chronyd conf failed, please check file /etc/chrony.conf", serverAddr)
+			}
+		}
+
+		// use internal ip to client chronyd server
+		for _, host := range runtime.GetAllHosts() {
+			if serverAddr == host.GetName() {
+				serverAddr = host.GetInternalAddress()
+				break
+			}
+		}
+
+		checkOrAddCmd := fmt.Sprintf(`grep -q '^server %s iburst' /etc/chrony.conf||sed '1a server %s iburst' -i /etc/chrony.conf`, serverAddr, serverAddr)
+		if _, err := runtime.GetRunner().SudoCmd(checkOrAddCmd, false); err != nil {
+			return errors.Wrapf(err, "set ntpserver: %s failed, please check file /etc/chrony.conf", serverAddr)
+		}
+	}
+
+	if len(n.KubeConf.Cluster.System.Timezone) > 0 {
+		setTimeZoneCmd := fmt.Sprintf("timedatectl set-timezone %s", n.KubeConf.Cluster.System.Timezone)
+		if _, err := runtime.GetRunner().SudoCmd(setTimeZoneCmd, false); err != nil {
+			return errors.Wrapf(err, "set timezone: %s failed", n.KubeConf.Cluster.System.Timezone)
+		}
+
+		if _, err := runtime.GetRunner().SudoCmd("timedatectl set-ntp true", false); err != nil {
+			return errors.Wrap(err, "timedatectl set-ntp true failed")
+		}
+	}
+
+	if len(n.KubeConf.Cluster.System.NtpServers) > 0 || len(n.KubeConf.Cluster.System.Timezone) > 0 {
```

Comment:
There is a `prepare` before this `action`, I think `len(n.KubeConf.Cluster.System.NtpServers) > 0` it can be removed